src/pages/Admin/Details/Form/DetailsForm.tsx

```diff
@@ -127,10 +130,26 @@ const DetailsForm: React.FC<DetailsFormProps> = ({ venue }) => {
 
         history.push(adminWorldSpacesUrl(venue?.worldId));
       } else {
+        if (!worldId) return;
+
+        const slugAlreadyExists = await checkSpaceExistsInWorld(
+          spaceSlug,
+          worldId
+        );
+
+        if (slugAlreadyExists) {
+          setError(
+            "name",
+            "manual",
+            "This space slug already exists in this world."
+          );
+          return;
+        }
+
```

Comment:
Same like above suggestion, don't use `setError()` let the throw happen, reuse the `error` returned from `useAsyncFn()` as a prop to `<SubmitError>`. A for can probably have multiple `<SubmitError>` and `<FormError>`.

Reply:
Fixed in: https://github.com/sparkletown/sparkle/pull/2548/commits/3d30910bba970e8e3052fe6b53a34775a7b5ed05